src/oatppjsonapi.cc

```diff
@@ -223,6 +223,33 @@ namespace dd
           }
       }
 
+    auto response = oatpp::web::protocol::http::outgoing::ResponseFactory::
+        createResponse(oatpp::web::protocol::http::Status(outcode, ""),
+                       stranswer.c_str());
+    response->putHeader(oatpp::web::protocol::http::Header::CONTENT_TYPE,
+                        "application/json");
+
+    std::string service;
+    if (janswer.HasMember("head") && janswer["head"].HasMember("service"))
+      service = janswer["head"]["service"].GetString();
+
+    return finalize_response(request, response, req_start_time, service);
+  }
+
+  std::shared_ptr<oatpp::web::protocol::http::outgoing::Response>
+  OatppJsonAPI::finalize_response(
+      const std::shared_ptr<
+          oatpp::web::server::api::ApiController::IncomingRequest> &request,
+      std::shared_ptr<oatpp::web::protocol::http::outgoing::Response>
+          &response,
+      const std::chrono::time_point<std::chrono::steady_clock> &req_start_time,
+      const std::string &service)
+  {
+    if (!FLAGS_allow_origin.empty())
+      response->putHeader("Access-Control-Allow-Origin",
```

Comment:
Just use the following:

```cpp
response->putHeader("Access-Control-Allow-Origin", FLAGS_allow_origin.c_str());
```

Also, note:
There is an out-of-the-box CORS support in oatpp:
- Per Endpoint with `ADD_CORS` macro
- And global - per connection handler - see [1.2.5 changelog](https://github.com/oatpp/oatpp/blob/master/changelog/1.2.5.md#enable-global-cors)